router/delivery/http/router_handler.go

```diff
@@ -380,10 +395,60 @@ func getValidTokenInTokenOutStr(c echo.Context) (tokenOutStr, tokenInStr string,
 		return "", "", errors.New("tokenOutDenom is required")
 	}
 
-	// Validate inpit denoms
+	// Validate input denoms
 	if err := domain.ValidateInputDenoms(tokenInStr, tokenOutStr); err != nil {
 		return "", "", err
 	}
 
 	return tokenOutStr, tokenInStr, nil
 }
+
+func getValidPoolID(c echo.Context) ([]uint64, error) {
+	// We accept two poolIDs and poolID parameters, and require at least one of them to be filled
+	poolIDStr := strings.Split(c.QueryParam("poolID"), ",")
+	if len(poolIDStr) == 0 {
+		return nil, errors.New("poolID is required")
+	}
+
+	var poolIDs []uint64
+	for _, v := range poolIDStr {
+		i, err := strconv.ParseUint(v, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		poolIDs = append(poolIDs, i)
+	}
+
+	return poolIDs, nil
+}
+
+func getPoolsValidTokenInTokensOut(c echo.Context) (poolID []uint64, tokenOut []string, tokenIn string, err error) {
```

Comment:
Could we add some unit tests for this?

I would like to make sure that we maintain backward compatibility. Detailed unit tests would help guarantee this and also make future changes simpler

Reply:
Tests were added, please let me know if you would like to include some additional test case.